src/main/kotlin/domain/Board.kt

```diff
@@ -4,5 +4,5 @@ class Board(
     private val matrix: Matrix,
```

Comment:
```suggestion
    matrix: Matrix,
```

matrix가 파라미터로만 사용되네요. private val을 없애줘도 좋을 것 같아요.